handbook/product/design/design-and-interaction-guidelines.md

```diff
@@ -0,0 +1,107 @@
+
+## Code pages design
+
+In Sourcegraph’s search and file experiences, code is the most important element. The design of these pages reflects that priority. Contrast is balanced so that code stands out above all else. Panels clearly segment content and tools from the code. Pages are full width, and margins and padding are slim to increase the density of information which helps us display as much code on the screen as possible. 
+
+The interaction model of the search experience is also unique. Clicking text on the page will almost always result in running a query or viewing a file. In a normal web application, we would clearly highlight these interactive elements so that the user understands where the actions are. If we were to do so on search pages, these elements would quickly outweigh and overwhelm the code. 
+
+### Examples:
+
+* [Search results](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:.*/go+auth&patternType=literal)
+* [File views](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/package.json)
+
+<img src='./margins-example.png' />
+
+_Small margins and reduced padding help to minimize scanning and scrolling. This allows developers to see code faster… and fit more of it on the screen._
+
+
+### Icons and labels
+
+In the search application, we generally display icons without labels. This is a common pattern in IDEs and other complicated UIs with many functions. This helps us maintain a high amount of information density while exposing many features within a single click. 
+
+<img src='./icons-example.png' />
+
+_Extensions and file menus compared with the VS Code’s git and windowing functions._
+
+## Web application page design
+
+We increasingly need to render standard webpages that present larger bodies of text to the user, accept user input, or allow management of elements. These pages require page widths and margins that promote text readability, a clear distinction between links and body copy, and padding and margins that allow elements to state their importance and guide users through the content. 
+
+
+### Example:
+
+[Repository groups](https://sourcegraph.com/refactor-python2-to-3)
+[Settings](https://sourcegraph.com/users/rrhyne/settings)
+[Extensions](https://sourcegraph.com/extensions)
+[Campaigns](https://k8s.sgdev.org/campaigns?visible=10)
+
+<img src='./signup-example.png' />
+
+_Clearly recognizable links help orient users to common signup actions._
+
+<img src='./large-margin-example.png' />
+
+_The large margins around the Kubernetes logo convey to the user: “This page is focused on Kubernetes”._
+
+### Web application icons
+
+In the web application sections, icons should always have an accompanying label. See this article on [Icon Usability](https://www.nngroup.com/articles/icon-usability/) from the Nielson Norman group for more information. Note that this stance is contradictory to the requirements for icons in search application pages.
```

Comment:
@rrhyne I wouldn't say it's ALWAYS necessary to have labels - for ex. 'info' icons with the tooltip on hover do not need labels. How about we change the wording to 'In the web application sections, icons should **generally** have an accompanying label.'?

Reply:
Good point.

Reply:
This should be resolved now, thanks for the suggestion.